### PR TITLE
Fix autosave when run on remote device

### LINKED ISSUE #5367: Missing auto save - Editor does not save project before runing on remote device.
**Operating system or device - Godot version:**
Windows 7, Godot 2.1 alpha, Android 4.4.2 arm

**Issue description** (what happened, and what was expected):
Editor save scene / script when user press "Play the project (F5)" or "Play the edited scene (F6)", so it is expected it will also save when run on remote device (click green android button), however it does not save before runing on remote device, which leads to confusion someone who is used to this feature. Personaly it took me fiew hairs and several minutes to discover why im typing and typing and whatever i could type - nothing changes in debug output - it just didnt saved when i ran the code)

**Steps to reproduce:**
Change something in script, run project by F5 key - editor will auto save scene / script then will run it.
Change something in script, run project on remote device by clicking on green android button - editor will NOT save your changes, and will run the old version all around untill you figure out it didnt saved.

### PR DESCRIPTION
**Pull request by volzhs**
Fix #5367

### CODE REVIEW

File: tools/editor/editor_node.cpp [hunk 1]
```diff
@@ -2639,11 +2639,16 @@ void EditorNode::_menu_option_confirm(int p_option,bool p_confirmed) {
...
 		case RUN_PLAY_NATIVE: {
-			_menu_option_confirm(RUN_STOP,true);
-			emit_signal("play_pressed");
-			editor_run.run_native_notify();
-
-
+			
+			bool autosave = EDITOR_DEF("run/auto_save_before_running",true);
+			if (autosave) {
+				_menu_option_confirm(FILE_SAVE_SCENE, false);
```
Comment (volzhs) [inline tools/editor/editor_node.cpp]: After #5421 merged, this also need to change to ``FILE_SAVE_ALL_SCENES``

**Comment by reduz** [comment]
Is this ok to merged or needs to be changed? (to this SAVE_ALL_SCENES option)

**Comment by volzhs** [comment]
I think #5421 should be merged first to make sure,
and then I can change it to ``SAVE_ALL_SCENES``

**Comment by reduz** [comment]
Having a generic IAP API based on plugins would be really good, it could work with Steam too

**Comment by reduz** [comment]
sorry wrong issue

**Pull request merged**